ocaml/typing/ctype.ml

```diff
@@ -5697,6 +5697,36 @@ let subtype_alloc_mode env trace a1 a2 =
   | Ok () -> ()
   | Error _ -> subtype_error ~env ~trace ~unification_trace:[]
 
+(* CR layouts v2.8: use the meet-with-constant morphism when available *)
+(* CR layouts v2.8: merge with Typecore.mode_cross_left when [Value] and
+   [Alloc] get unified *)
+(* The approach here works only for 2-element modal axes. *)
+let mode_cross_left env ty mode =
```

Comment:
Any chance we might benefit from a principality check here?

Reply:
I believe there is a principality check here.

Reply:
Do you mean there is no principality check here? I think you are right - the original PR(https://github.com/ocaml-flambda/flambda-backend/pull/1701) didn't check for principality for certain reason.